### PR TITLE
Added splitter function

### PR DESCRIPTION
**Pull request by N8python**
splitter

A function that takes a string or number as a divider, and a series of strings to print out, divided. If the first parameter is a number, it looks it up in self.dividers. Otherwise, it uses the string provided as a divider. If only one additional string is provided, no dividers are returned.

### CODE REVIEW

**Comment by RDIL** [comment]
First, I’m confused.  Second, please fix the broken status checks.

**Comment by N8python** [comment]
What do you mean? There is nothing wrong with the code.

File: area4/__init__.py [hunk 1]
```diff
@@ -65,7 +65,16 @@ def divider(self, number) -> str:
                 return self.dividers[number].split("\n")[0]
             except IndexError:
                 raise ValueError('That divider doesn\'t exist!')
-
+    def logDivider(self, div, *stuff) -> None:
```
Comment (RDIL) [inline area4/__init__.py]: Needs DocStrings/PyDoc

File: area4/__init__.py [hunk 2]
```diff
@@ -65,7 +65,16 @@ def divider(self, number) -> str:
...
+        if(type(div) is int) div = self.dividers[div]
```
Comment (RDIL) [inline area4/__init__.py]: This isn’t a thing in python.

File: area4/__init__.py [hunk 3]
```diff
@@ -65,7 +65,16 @@ def divider(self, number) -> str:
...
+        if len(stuff) == 1:
+            print(stuff[0])
+            return
+        for s in stuff:
+           print(s)
```
Comment (RDIL) [inline area4/__init__.py]: Please avoid using print, return is preferred. If you must, make sure you add a return type annotation of None.

**Comment by RDIL** [comment]
@N8python see requested changes above.

File: area4/__init__.py (hunk 1 above)
Comment (N8python) [reply area4/__init__.py]: Could you add that? I'm not sure if I could do it right.

File: area4/__init__.py (hunk 3 above)
Comment (N8python) [reply area4/__init__.py]: I did!

File: area4/__init__.py (hunk 1 above)
Comment (RDIL) [reply area4/__init__.py]: Sure

**Comment by N8python** [comment]
I made it return a string!

File: area4/__init__.py [hunk 4]
```diff
@@ -65,7 +65,20 @@ def divider(self, number) -> str:
                 return self.dividers[number].split("\n")[0]
             except IndexError:
                 raise ValueError('That divider doesn\'t exist!')
-
+    def logDivider(self, div, *stuff) -> str:
+        retstr = "";
+        if type(div) is int: div = self.dividers[div]
+        if len(stuff) == 1:
+            retstr = stuff[0]
+            return retstr
```
Comment (CoffeeGoddess) [inline area4/__init__.py]:
```suggestion
            return stuff[0]
```

File: area4/__init__.py (hunk 2 above)
Comment (N8python) [reply area4/__init__.py]: Oh - sorry. I was in "js mode".

File: area4/__init__.py [hunk 5]
```diff
@@ -65,7 +65,20 @@ def divider(self, number) -> str:
                 return self.dividers[number].split("\n")[0]
             except IndexError:
                 raise ValueError('That divider doesn\'t exist!')
-
+    def logDivider(self, div, *stuff) -> str:
+        retstr = "";
+        if type(div) is int: div = self.dividers[div]
+        if len(stuff) == 1:
+            retstr = stuff[0]
```
Comment (CoffeeGoddess) [inline area4/__init__.py]:
```suggestion

```

Comment (N8python) [reply area4/__init__.py]: I don't think we should. We want to return the only string given if you simply pass logDivider('', "Foo")

File: area4/__init__.py (hunk 2 above)
Comment (RDIL) [reply area4/__init__.py]: Haha no problem

File: area4/__init__.py (hunk 5 above)
Comment (RDIL) [reply area4/__init__.py]: What coffee is trying to say is that this is a simplified method.

File: area4/__init__.py [hunk 6]
```diff
@@ -65,7 +65,33 @@ def divider(self, number) -> str:
                 return self.dividers[number].split("\n")[0]
             except IndexError:
                 raise ValueError('That divider doesn\'t exist!')
+    
+    
+    def logDivider(self, div, *stuff) -> str:
+        """
```
Comment (RDIL) [inline area4/__init__.py]: @N8python can you review the documentation I just put in and see if it is accurate?

**Comment by RDIL** [comment]
@N8python in addition, the following issues need to be fixed before merge (@rocketpropelledpotato please don't merge until me and Coffee approve the PR):
```
./area4/__init__.py:68:1: W293 blank line contains whitespace
./area4/__init__.py:69:1: W293 blank line contains whitespace
./area4/__init__.py:70:5: E303 too many blank lines (2)
./area4/__init__.py:72:80: E501 line too long (106 > 79 characters)
./area4/__init__.py:83:28: E701 multiple statements on one line (colon)
./area4/__init__.py:88:12: E111 indentation is not a multiple of four
./area4/__init__.py:88:18: E225 missing whitespace around operator
./area4/__init__.py:89:12: E111 indentation is not a multiple of four
./area4/__init__.py:89:18: E225 missing whitespace around operator
./area4/__init__.py:90:12: E111 indentation is not a multiple of four
./area4/__init__.py:90:18: E225 missing whitespace around operator
./area4/__init__.py:91:12: E111 indentation is not a multiple of four
./area4/__init__.py:91:18: E225 missing whitespace around operator
./area4/__init__.py:94:1: W293 blank line contains whitespace
./area4/__init__.py:95:5: E303 too many blank lines (2)
./area4/__init__.py:96:5: E301 expected 1 blank line, found 0
```

File: area4/__init__.py (hunk 5 above)
Comment (N8python) [reply area4/__init__.py]: Oh, got it.

**Comment by N8python** [comment]
Can you guys approve, and we go ahead and merge?

**Comment by RDIL** [comment]
`flake8`, `pycodestyle`, and `pydocstyle` are required @N8python.  They must be passing before we can approve and merge.

**Comment by N8python** [comment]
Could you help me do what they want?

**Comment by RDIL** [comment]
@N8python sure, if you click `Details` you can see all the issues.

**Comment by N8python** [comment]
Cool!

**Comment by RDIL** [comment]
It seems that there are no runtime issues with the code, only style issues.

**Comment by N8python** [comment]
Got it. Before you guys came in and styled it, I was programming in a JS style. Heh.

**Comment by RDIL** [comment]
Yeah, we follow PEP8 here.  Its a very interesting guide, I suggest you give it a read:  
https://www.python.org/dev/peps/pep-0008/  
;)

**Comment by RDIL** [comment]
One last issue.  Also the function needs documentation.

**Comment by N8python** [comment]
I thought you added the documentation.

File: area4/__init__.py (hunk 6 above)
Comment (N8python) [reply area4/__init__.py]: It's accurate.

**Comment by N8python** [comment]
Could you work on some of that RDIL? I have to go now, and you could probably do it better than me.

**Comment by RDIL** [comment]
Coffee or I can do the documentation, but you will have to write tests as this doesn't really make sense to me.  @CoffeeGoddess tag for docs and @rocketpropelledpotato could you look over his python please.

**Comment by N8python** [comment]
Got it!

**Comment by RDIL** [comment]
@N8python all you have to do now is edit `test-code.py`.  What you need to do is add a new function that compares the outputted code to a possible output.  For example:  
```python
def test() -> None:
    print("[DEBUG] Running logDivider tests")
    if not logDivider(some, stuff) == what_it_should_return:
        raise RuntimeError("Failed to run test")
    else:
        print("it works yay")
```
then call the function at the end of the file

**Comment by N8python** [comment]
Got it

**Comment by RDIL** [comment]
Don't worry about the CodeTests failing - that will be fixed soon.  For now, please fix the flake8 issues

**Comment by N8python** [comment]
Ok

**Comment by RDIL** [comment]
The PR is ready but we can’t merge until after we release this stable.

**Comment by N8python** [comment]
What do you mean by “stable”?

**Comment by RDIL** [comment]
Right now where the master branch is, we are confident that the code works and have already drafted changelogs for the release.   2.0.1 should be deployed some time this week, then this PR can be deployed in 2.0.2 next week.

**Comment by N8python** [comment]
Got it! Thx for helping with the pull request!

**Comment by RDIL** [comment]
Wait no, scratch that, I’ll deploy 2.0.1 tonight and 2.0.2 on Friday

**Comment by N8python** [comment]
Got it!

**Comment by RDIL** [comment]
In the mean time, feel free to open more PRs.

**Comment by RDIL** [comment]
I fixed tests so now it doesn't use an old distribution.

**Comment by RDIL** [comment]
@N8python the test you wrote failed, please take a look.

**Comment by RDIL** [comment]
@N8python please fix tests

**Comment by N8python** [comment]
I'm not sure what is wrong. Could you take a look at it?

**Comment by RDIL** [comment]
@N8python so I took a look and it turns out that the *issue is with the `test-code.py` file*.  The first test **condition1 is working but condition2 is not matching**.  Take a look.  **I will be on an airplane for the next few hours so I cannot help.**

**Comment by N8python** [comment]
When I have time, I  will.

**Comment by RDIL** [comment]
@N8python after some careful analysis the following statement returns `False`:
`d.logDivider("---", "Hello", "Hello") == "Hello\n---\nHello"`

**Comment by N8python** [comment]
I saw. I don’t know why though

**Comment by RDIL** [comment]
Try just remaking the line from scratch.

**Comment by N8python** [comment]
Sure - I will within the next few hours.

**Comment by RDIL** [comment]
Please try to take a look now.

**Comment by RDIL** [comment]
Tests should be fixed but CODEOWNERS are on vacation

**Comment by N8python** [comment]
Got it.

**Comment by RDIL** [comment]
Need approval from @rocketpropelledpotato or @CoffeeGoddess

**Comment by RDIL** [comment]
This all looks good.  Just I think we should rename it something better.

File: area4/__init__.py [hunk 7]
```diff
@@ -64,6 +64,27 @@ def divider(self, number):
...
+
+        :return: newly made value
+        :rtype: str
+        :param self:
+        :param div: the divider
+        """
+        retstr = ""
+        if type(div) is int:
+            div = self.dividers[div]
+        if len(args) == 1:
```
Comment (RDIL) [inline area4/__init__.py]: Also can we change this to `elif` without error?

Comment (RDIL) [reply area4/__init__.py]:
```suggestion
        elif len(args) == 1:
```
?

**Comment by RDIL** [comment]
@N8python I think this is finally ready for merge.

**Comment by N8python** [comment]
I don't think so. I've been working on some other stuff. The function
dosen't work completetly.

**Comment by RDIL** [comment]
What do you mean?

**Comment by N8python** [comment]
OOP! Never mind, it works perfectly. I was just testing it wrong.
Sorry about that.

**Comment by N8python** [comment]
It works fine - I was testing it wrong. Sorry for raising a false alarm.

**Comment by RDIL** [comment]
Ready?

**Comment by N8python** [comment]
Let's do this. Thanks for helping my splitter function idea become a
reality.

**Comment by RDIL** [comment]
:)  
Also feel free to contribute anything else 🥇 !

**Pull request merged**